### PR TITLE
feat: add dcs instances data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dcs instances data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dcs instances data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dcs/' TESTARGS='-run TestAccDatasourceDcsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs/ -v -run TestAccDatasourceDcsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDcsInstance_basic 
=== PAUSE TestAccDatasourceDcsInstance_basic 
=== CONT  TestAccDatasourceDcsInstance_basic 
--- PASS: TestAccDatasourceDcsInstance_basic (80.54s) 
PASS                                                                                                            
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       80.592s
```

### CODE REVIEW

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+		},
+	}
+	return &sc
+}
+
+func resourceDcsInstanceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	var mErr *multierror.Error
```
Comment (deer-hang) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go:216]: remove this line to line:270

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+		capacityMinor := utils.PathSearch("capacity", v, nil)
+		if capacity == 0 {
+			capacity, _ = strconv.ParseFloat(capacityMinor.(string), floatBitSize)
+		}
+
+		securityGroupID := utils.PathSearch("security_group_id", v, nil)
+		// If security_group_id is not set, the default value is returned: securityGroupId. Bad design.
+		if securityGroupID == "securityGroupId" {
+			securityGroupID = ""
+		}
```
Comment (deer-hang) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go:288]: the description "If security_group_id is not set, the default value is returned: securityGroupId. Bad design" not match the code: 
if securityGroupID == "securityGroupId" {
			securityGroupID = ""
}
please check it.

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go:216]: the code is automatically generated, the style is unified

File: docs/data-sources/dcs_instances.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+# huaweicloud_dcs_instances
+
+Use this data source to get the list of DCS instances.
+
+## Example Usage
+
+```hcl
+data "huaweicloud_dcs_instances" "test" {
+  name          = "test_name"
+  status        = "RUNNING"
```
Comment (ShiChangkuo) [inline docs/data-sources/dcs_instances.md]: extra spaces

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the domain name of the instance.`,
+			},
+			"user_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the user ID.`,
+			},
+			"user_name": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go]: `user_id` and `user_name` are unnecessary, you can delete them

**Pull request merged**